shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -461,19 +491,48 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
   }
 }
 
+- (void)propagateKeyEvent:(NSEvent*)event {
+  for (NSResponder* responder in self.additionalKeyResponders) {
+    if (event.type == NSEventTypeKeyDown) {
+      if ([responder respondsToSelector:@selector(keyDown:)]) {
+        [responder keyDown:event];
+      }
+    } else if (event.type == NSEventTypeKeyUp) {
+      if ([responder respondsToSelector:@selector(keyUp:)]) {
+        [responder keyUp:event];
+      }
+    }
+  }
```

Comment:
I missed this before; we should be passing along to the `nextResponder` if the code above doesn't handle it. As written this isn't actually propagating the event. (additionalKeyResponders is, in practice, just the text input plugin.)

Reply:
Oh! I wondered about that, but I forgot to circle back and investigate. OK, I'll do that.

Reply:
OK, done.  I also noticed that `additionalKeyResponders` wasn't actually removing responders in `removeKeyResponder`, so we were accumulating them, and I wonder what bugs we could have seen from that. It now removes them.

Reply:
Oops! I wonder how that happened.